src/java/org/apache/cassandra/metrics/ClientMetrics.java

```diff
@@ -51,20 +93,56 @@ public final class ClientMetrics
     private Meter protocolException;
     private Meter unknownException;
 
+    private static final String AUTH_SUCCESS = "AuthSuccess";
+
+    private static final String AUTH_FAILURE = "AuthFailure";
+
+    private static final String CONNECTED_NATIVE_CLIENTS = "ConnectedNativeClients";
+
     private ClientMetrics()
     {
     }
 
+    /**
+     * @deprecated by {@link #markAuthSuccess(String)}
+     */
+    @Deprecated(since="5.1", forRemoval = true)
     public void markAuthSuccess()
     {
-        authSuccess.mark();
+        markAuthSuccess(null);
     }
 
+    public void markAuthSuccess(String mode)
+    {
+        markAuthMeter(authSuccess, authSuccessByMode, mode);
+    }
+
+    /**
+     * @deprecated by {@link #markAuthFailure(String)}
+     */
+    @Deprecated(since="5.1", forRemoval = true)
     public void markAuthFailure()
     {
         authFailure.mark();
     }
 
+    public void markAuthFailure(String mode)
+    {
+        markAuthMeter(authFailure, authFailureByMode, mode);
+    }
+
+    private void markAuthMeter(Meter meter, Map<String, Meter> meterMap, String mode)
```

Comment:
I would personally get rid of this method and just put it into methods which call this logic. The only variable there would be meterMap which would be either `authFailureByMode` or `authSuccessByMode`. This kind of repetition is not a big deal.

Reply:
:+1: yeah at this point the method has been reduced enough that it doesn't offer much, i'll factor that out